src/coupling.jl

```diff
@@ -0,0 +1,68 @@
+SplitCoupledJumpProblem(prob::AbstractJumpProblem,prob_control::AbstractJumpProblem,aggregator::AbstractAggregatorAlgorithm,coupling_map::Vector{Tuple{Int64,Int64}})= JumpProblem(cat_problems(prob,prob_control),aggregator,build_split_jumps(prob,prob_control,coupling_map)...)
+
+# make new discrete problem by joining initial_data
+function cat_problems(prob::AbstractJumpProblem,prob_control::AbstractJumpProblem)
+  coupled_u0 = cat(1,prob.prob.u0,prob_control.prob.u0)
+  DiscreteProblem(coupled_u0,prob.prob.tspan)
+end
+
+function build_split_jumps(prob::AbstractJumpProblem,prob_control::AbstractJumpProblem,coupling_map::Vector{Tuple{Int64,Int64}})
+  num_jumps = length(prob.discrete_jump_aggregation.rates)
+  d = length(prob.prob.u0)
+  jumps = []
+  uncoupled = deleteat!(Vector(1:num_jumps),[c[1] for c in coupling_map ])
```

Comment:
You shouldn't have to create the whole vector and deleteat. This is likely much worse than building by pushing (which is automatically amortized), especially since the comprehension forms an extra vector here as well. But there's likely a way to do it directly.